src/App.tsx

```diff
@@ -1,29 +1,54 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const items = getNumbers(1, 42).map(n => +n);
+const total = 42;
+const options = [3, 5, 10, 20];
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(options[1]);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const perPageSelector = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(+event.target.value);
+    setCurrentPage(1);
+  };
+
+  function getItemsOnPage() {
+    return items.filter(item => {
+      return (
+        item > currentPage * perPage - perPage && item <= currentPage * perPage
+      );
+    });
+  }
+
+  const itemsOnPage = getItemsOnPage();
```

Comment:
and if you're a familiar with `useMemo` hook it'd nice to use it here
P.S. I've written in one line, but you can change to pass linter requirements
```suggestion
  const itemsOnPage = items.filter(item => item > currentPage * perPage - perPage && item <= currentPage * perPage);
```